source/clear-linux/guides/maintenance/autospec.rst

```diff
@@ -197,49 +251,137 @@ a new number that is higher than the |CL| OS version, and rebuild the package.
+
+This example shows how to modify control files to correct build failures that
+autospec is unable to resolve. In this example you will add a missing license
+and dependencies in order for autospec to complete a successful build.
+
+Complete `Example 1: Complete setup for building source`_ before starting this
+example.
 
-#. At :file:`~/clearlinux/packages/dmidecode`, build the modified
-   ``dmidecode`` package:
+#. Navigate to the autospec workspace:
+
+   .. code-block:: bash
+
+      cd ~/clearlinux
+
+#. Build the opae-sdk RPM:
+
+   .. code-block:: bash
+
+      make autospecnew URL="https://github.com/OPAE/opae-sdk/archive/0.13.0.tar.gz" NAME="opae-sdk"
+
+   This will give an error for a missing license file:
+
+   .. code-block:: console
+
+      [FATAL]    Cannot find any license or opae-sdk.license file!
+
+#. Navigate to the package with build failures:
+
+   .. code-block:: bash
+
+      cd packages/opae-sdk
+
+#. Add a license:
+
+   .. code-block:: bash
+
+      echo "MIT" > opae-sdk.license
+
+#. Run autospec again:
 
    .. code-block:: bash
 
       make autospec
 
-   When the process completes, you will see new RPM packages in the
-   :file:`results/` folder.
+   This will result in a generic error:
+
+   .. code-block:: console
+
+      [FATAL]    Build failed, aborting
 
-#. To view the new RPM packages, enter:
+#. Open the build log to view the error details:
 
    .. code-block:: bash
 
-      ls /clearlinux/packages/dmidecode/results/
+      cat ./results/build.log
 
-**Congratulations!**
+   In the build log, you will find details for the specific failures. In this
+   instance, there are missing dependencies:
 
-You've successfully created a RPM.
+   .. code-block:: console
 
-Next steps
-**********
+      CMake Error: The following variables are used in this project, but they are set to NOTFOUND.  Please set them or make sure they are set and tested correctly in the CMake files:
+      CJSON_LIBRARY
+         linked by target "opae-c++-utils" in directory /builddir/build/BUILD/opae-sdk-0.13.0/tools/c++utilslib
+      json-c_LIBRARIES
+         linked by target "opae-c" in directory /builddir/build/BUILD/opae-sdk-0.13.0/libopae
+      libuuid_LIBRARIES
+         linked by target "opae-c" in directory /builddir/build/BUILD/opae-sdk-0.13.0/libopae
+
+#. Search for the json-c package in |CL|:
+
+   .. code-block:: bash
 
-Now you can create a custom bundle with your new RPM and use it with |CL|:
+      grep 'json-c\.so$' ~/clearlinux/packages/*/*.spec
 
-* Use the :ref:`Mixer tool <mixer>` to add a new bundle to your derivative of |CL|.
-* Use the :ref:`Mixin tool <mixin>` to customize your upstream |CL| installation with a new bundle.
+#. Add the json-c-dev dependency:
+
+   .. code-block:: bash
+
+      echo "json-c-dev" >> buildreq_add
+
+#. Search for the libuuid package in |CL| and add the util-linux-dev dependency:
+
+   .. code-block:: bash
+
+      grep 'libuuid\.so$' ~/clearlinux/packages/*/*.spec
```

Comment:
Please note that this only works if the user cloned all of the upstream package repos.